deploy/local_single_user_deploy.sh

```diff
@@ -38,40 +39,50 @@ clean=false
 EXEC_MODE_COUNT=0
 
 # Helper Methods _______________________________________________________________________________________________________
-check_dependencies() {
-    local dependencies=("jq" "ldd" "nohup" "pkill" "readlink")
+start_service() {
+    local bin="$1"
+    local args="$2"
+    local monitor_file="$3"
+    echo -e "${DEBUG}Launching $bin $args ...${NC}"
+    LD_LIBRARY_PATH=${LIB_DIR} nohup ${bin} ${args} > ${MONITOR_DIR}/${monitor_file} 2>&1 &
+}
 
-    echo -e "${DEBUG}Checking required dependencies...${NC}"
-    for dep in "${dependencies[@]}"; do
-        if ! command -v $dep &> /dev/null; then
-            echo -e "${ERR}Dependency $dep is not installed. Please install it and try again.${NC}"
-            exit 1
-        fi
-    done
-    echo -e "${DEBUG}All required dependencies are installed.${NC}"
+kill_service() {
+    local bin="$1"
+    echo -e "${DEBUG}Killing $(basename ${bin}) ...${NC}"
+    pkill -9 -f ${bin}
 }
 
-check_directories() {
-    echo -e "${DEBUG}Checking required directories...${NC}"
-    local directories=("${WORK_DIR}" "${LIB_DIR}" "${CONF_DIR}" "${BIN_DIR}")
+stop_service() {
+    local bin="$1"
+    local timeout="$2"
 
-    for dir in "${directories[@]}"; do
-        if [[ ! -d ${dir} ]]; then
-            echo -e "${ERR}Directory ${dir} does not exist. Please create it and try again.${NC}"
-            exit 1
+    # Stop all processes of the service in parallel
+    local start_time=$(date +%s)
+    echo -e "${DEBUG}Stopping $(basename ${bin}) ...${NC}"
+    pkill -f ${bin}
+
+    # Wait for processes to stop with a timeout
+    while true; do
+        if pgrep -f "${bin}" >/dev/null; then
```

Comment:
Found a bug. 
If `work-dir` is passed differently with and without trailing `/` in start and stop operation, `pgrep -f` fails to find the running processes. I guess the same thing for `pkill -f`. Maybe consider passing `basename binary` to `pgrep -f` and `pkill -f`.